torch/csrc/jit/tensorexpr/ir_simplifier.h

```diff
@@ -9,17 +11,260 @@ namespace torch {
 namespace jit {
 namespace tensorexpr {
 
```

Comment:
One other thing: there is a lot of code here and it is not-trivial. It might be worth adding a high-level comment here to explain what it does and roughly how it does what it does.